pkg/runtime/interfaces.go

```diff
@@ -30,12 +30,23 @@ const (
 	APIVersionInternal = "__internal"
 )
 
+// GroupVersioner conveys information about a desired target version for objects being converted.
+type GroupVersioner interface {
```

Comment:
The problem is that responsibility today belongs to the scheme, which has the info "what version *may* this kind be converted to".  This type is "given this info, do you you want to express a preference about *where* to go".